h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -221,6 +227,16 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(v1.getString());
+            ValueArray va = (ValueArray) v0.convertTo(TypeInfo.TYPE_ARRAY);
+            for (Value v : va.getList()) {
+                if (v != null && !(v instanceof ValueNull)) {
```

Comment:
1. Why `v != null` is needed here?

2. Use `v == ValueNull.INSTANCE` for a null check.

3. If an optional third argument is specified, it should be used as replacement for null values.

Reply:
fixed null check and added 3rd arg replacement